syz-manager/hub.go

```diff
@@ -73,19 +73,28 @@ type HubConnector struct {
 type HubManagerView interface {
 	getMinimizedCorpus() (corpus, repros [][]byte)
 	addNewCandidates(candidates []rpctype.Candidate)
+	hubIsUnreachable()
 }
 
+const maxAttempts = 3
+
 func (hc *HubConnector) loop() {
 	var hub *rpctype.RPCClient
-	for ; ; time.Sleep(10 * time.Minute) {
+	var connectedOnce bool
+	for attempt := 0; ; time.Sleep(10 * time.Minute) {
 		corpus, repros := hc.mgr.getMinimizedCorpus()
 		hc.newRepros = append(hc.newRepros, repros...)
 		if hub == nil {
+			attempt++
 			var err error
 			if hub, err = hc.connect(corpus); err != nil {
 				log.Logf(0, "failed to connect to hub at %v: %v", hc.cfg.HubAddr, err)
+				if attempt >= maxAttempts && !connectedOnce {
+					break
+				}
 				continue
 			}
+			connectedOnce = true
```

Comment:
We need to set it when Sync succeeded. It promoted the status.
It's also better to keep trying to connect.
I would add to be beginning of the loop:
```
				const maxAttempts = 3
				if attempt == maxAttempts && !connectedOnce {
					hc.mgr.hubIsUnreachable()
				}
```

Reply:
PTAL